src/utilities/types.tsx

```diff
@@ -0,0 +1,19 @@
+export interface Items {
+  id: string;
+  itemName: string;
+  rarity: string;
+  description: string;
+  stackType: string;
+}
+
+export interface Survivor {
+  id: string;
+  name: string;
+  imageLink: string;
+}
+
+export interface UserSelection {
+  userID: number;
+  userSurvivor: Survivor;
+  userItems: Items[];
+}
```

Comment:
possible to have strong types for some of these?

ie; if theres only 3 rarities (common, rare, legendary)

```
enum itemRarities {
    COMMON = "COMMON",
    RARE = "RARE",
    LEGENDARY = "LEGENDARY",
}
```

which then allows you to do

`rarity: keyof typeof itemRarities;`

Reply:
probably can be done for 
- rarity
- stackType

Reply:
![image](https://github.com/JeffreyChuCPA/Risk-of-Rain-2-Multishop-Simulator/assets/141854177/e820418f-5dfe-46fb-8533-da6f5bf12bdc)
![image](https://github.com/JeffreyChuCPA/Risk-of-Rain-2-Multishop-Simulator/assets/141854177/2f2c2654-e262-4298-a921-6697445839b4)
unsure what you mean by this. Are you saying to explicitly define "rarity" to be 1 of 3 string options? I attached a sample of what the received data would be for common and uncommon items. Unsure what you mean by this part also : 
![image](https://github.com/JeffreyChuCPA/Risk-of-Rain-2-Multishop-Simulator/assets/141854177/51d4c396-6802-4c5a-a58d-dc0521d81ee3)

Reply:
so, the general idea is that `rarity` can only be a finite number of values (there are only 5-6 rarities in the game). thus, we can do even stronger typing than just `string`

```
enum itemRarities {
    COMMON = "COMMON",
    RARE = "RARE",
    LEGENDARY = "LEGENDARY",
}

export interface Items {
  id: string;
  itemName: string;
  rarity: keyof typeof itemRarities;
  description: string;
  stackType: string;
}
```